OpenRA.Mods.Common/Traits/UpgradesIcons.cs

```diff
@@ -0,0 +1,137 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Attach this to the player actor. When attached, enables all actors possessing the UpgradesWhenCreated ",
+		"trait to have their production queue icons render with an overlay defined in this trait. ",
+		"The icon change occurs when UpgradesWhenCreated.Prerequisites are met.")]
+	public class UpgradesIconsInfo : ITraitInfo, Requires<TechTreeInfo>
+	{
+		[Desc("Image used for this decoration. Defaults to the actor's type.")]
+		public readonly string Image = "rank";
+
+		[Desc("Sequence used for this decoration (can be animated).")]
+		public readonly string Sequence = "rank";
+
+		[Desc("Palette to render the sprite in. Reference the world actor's PaletteFrom* traits.")]
+		public readonly string Palette = "effect";
+
+		[Desc("Point in the actor's bounding box used as reference for offsetting the decoration image." +
+			"Possible values are any combination of Top, VCenter, Bottom and Left, HCenter, Right separated by a comma.")]
+		public readonly ReferencePoints ReferencePoint = ReferencePoints.Top | ReferencePoints.Right;
+
+		[Desc("Pixel offset relative to the actor's bounding box' reference point.")]
+		public readonly int2 Offset = new int2(-7, 6);
```

Comment:
It won't go away, as stated. Sane defaults for a single mod are better than no default values.